### PR TITLE
Question navigation

### PR DESCRIPTION
**Pull request by barbarakos**
Implemented switching to new question and coloring progress bar view for each correct/incorrect answer.
Simulator Screen Shot - iPhone 14 Plus - 2022-11-08 at 12 23 40

### CODE REVIEW

File: QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift [hunk 1]
```diff
@@ -33,11 +33,19 @@ class QuizSessionViewController: UIViewController {
...
     func nextQuestion() {
-        viewModel.nextQuestion()
+        if viewModel.currentQuestion.index+1 < viewModel.questions.count {
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+                self?.viewModel.nextQuestion()
+            }
+        } else {
+            // show quiz results
+        }
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift]: This should be done in viewModel

File: QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift [hunk 2]
```diff
@@ -137,6 +145,23 @@ extension QuizSessionViewController: ConstructViewsProtocol {
...
+    func bindViews() {
+        questionView
+            .$isCorrectAnswer
+            .sink {  [weak self] isCorrect in
+                guard let self = self, let isCorrect = isCorrect else { return }
+
+                self.colorProgressViews(isCorrect: isCorrect)
+                self.nextQuestion()
+            }
+            .store(in: &cancellables)
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift]: what happens when the user answered incorrectly?

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

File: QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift (hunk 2 above)
Comment (barbarakos) [reply QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift]: If the user answers the question incorrectly then in the ` ` ` colorProgressViews()` ` `  function colors the top progress view that represents that question red.

File: QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift [hunk 3]
```diff
@@ -137,6 +145,23 @@ extension QuizSessionViewController: ConstructViewsProtocol {
 // MARK: ProgressBar functions
 extension QuizSessionViewController {
 
+    func bindViews() {
+        questionView
+            .$isCorrectAnswer
+            .sink {  [weak self] isCorrect in
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewControllers/QuizSessionViewController.swift]: Use compactMap to remove nil values

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

**Review by dorastipkovic — APPROVED** [review]

**Pull request merged**